lib/logstash/plugin_mixins/event_support/event_factory.rb

```diff
@@ -0,0 +1,42 @@
+require 'logstash/event'
+
+module LogStash; module PluginMixins; module EventSupport
+
+  class EventFactory
+
+    INSTANCE = new
+
+    def new_event(data = {})
+      event = LogStash::Event.new(data)
+      normalize(event, data)
+      event
+    end
+
+    def normalize(event, data = nil)
```

Comment:
we'd like to potentially adjust created events that bypass the event factory (again due JSON codec)

Reply:
UPDATE: we can decouple from `Event.from_json` so this normalization part should be dropped.